checkpoint/recovery.go

```diff
@@ -169,22 +168,35 @@ func RecoverWithDb(
 	if err != nil {
 		return nil, err
 	}
-	return recoverFromLocalFile(ctx, logger, db, localDB, fs, cfg, cpFile)
+	var newDB *sql.Database
+	createDb := func() (*sql.Database, error) {
+		var err error
+		newDB, err = sql.Open("file:" + filepath.Join(cfg.DataDir, cfg.DbFile))
+		return newDB, err
+	}
+	defer func() {
+		if newDB != nil {
+			newDB.Close()
+		}
+	}()
```

Comment:
Please no...

As far as I can tell this only exists such that you can pass an in-memory DB to a test. Instead set the `DataDir` to `t.TempDir()` and `DbFile` to a name of your liking and open the file after the recovery for assertion.